sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -367,7 +369,7 @@ def set_configuration_setting(
                 key=key_value.key,  # type: ignore
                 label=key_value.label,
                 if_match=prep_if_match(configuration_setting.etag, match_condition),
-                if_none_match=prep_if_none_match(configuration_setting.etag, match_condition),
+                if_none_match=prep_if_none_match(etag or configuration_setting.etag, match_condition),
```

Comment:
Might make sense to add in the docstring that `etag` value takes precedence over etag set in the configuration_setting

Reply:
Updated the docstring